### ISSUE TITLE
Installing features should be first

### ISSUE THREAD

**Issue by dave08** [micronaut-projects/micronaut-kotlin#17]
Currently installing features is done through `KtorApplicationBuilder` just like the routes, but certain route initialization depends on features already installed (such as Authentication). Since DI is done with no particular order, this could cause failures in initialization. This problem has currently caused me to make one big `KtorApplicationBuilder` for the whole initialization (which is pretty messy). I suggest an `KtorApplicationInitializerBuilder` or something that runs before the other modules.

**Comment by edrd-f** [comment]
Hi @dave08. There's now a separate builder for routing, which is loaded after application builders. This should solve the issue. Can you take a look and confirm?

**Comment by dave08** [comment]
Thanks for the fix! Do you have a release I can reference in my gradle dependencies @edrd-f?

**Comment by edrd-f** [comment]
Hey @dave08, glad to contribute. 🙂  

You can use the build snapshot version:

```kotlin
implementation("io.micronaut.kotlin:micronaut-ktor:1.0.0.BUILD-SNAPSHOT")
```
(https://oss.sonatype.org/#nexus-search;classname~KtorRoutingBuilder)

You will also need to include the Sonatype repository, if you didn't already:

```
maven("http://oss.sonatype.org/content/repositories/snapshots/")
```

**Comment by graemerocher** [comment]
I have released 1.0.0.M2 so you can give that version a go if you don't want to use the snapshots
